src/nvim/eval.c

```diff
@@ -16801,17 +16815,17 @@ static void f_type(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   int n = -1;
 
   switch (argvars[0].v_type) {
-    case VAR_NUMBER: n = 0; break;
-    case VAR_STRING: n = 1; break;
-    case VAR_FUNC:   n = 2; break;
-    case VAR_LIST:   n = 3; break;
-    case VAR_DICT:   n = 4; break;
-    case VAR_FLOAT:  n = 5; break;
+    case VAR_NUMBER: n = VAR_TYPE_NUMBER; break;
+    case VAR_STRING: n = VAR_TYPE_STRING; break;
+    case VAR_FUNC:   n = VAR_TYPE_FUNC; break;
+    case VAR_LIST:   n = VAR_TYPE_LIST; break;
+    case VAR_DICT:   n = VAR_TYPE_DICT; break;
+    case VAR_FLOAT:  n = VAR_TYPE_FLOAT; break;
     case VAR_SPECIAL: {
       switch (argvars[0].vval.v_special) {
         case kSpecialVarTrue:
         case kSpecialVarFalse: {
-          n = 6;
+          n = VAR_TYPE_BOOL;
           break;
         }
         case kSpecialVarNull: {
```

Comment:
I see here that `:echo type(v:null)` returns `7`. In Vim8,

    :echo type(v:null) " returns 7
    :echo type(v:none) " returns 7

So maybe we need `v:t_none` after all. (I still don't think it would make sense to introduce `v:t_null`, though).

Though IMO we should just wait until someone actually ever requests this. It's a silly feature IMO.